packages/checkbox/src/useCheckbox.ts

```diff
@@ -1,23 +1,183 @@
     return (elFormItem || {} as any).elFormItemSize
   })
   return {
     isGroup,
-    focus,
-    _checkboxGroup,
+    checkboxGroup,
     elForm,
     ELEMENT,
-    _elFormItemSize,
+    elFormItemSize,
     elFormItem,
   }
 }
+
+const useModel = (props: ICheckboxProps) => {
+  let selfModel = false
+  const { emit } = getCurrentInstance()
+  const { isGroup, checkboxGroup } = useCheckboxGroup()
+  const isLimitExceeded = ref(false)
+  const store = computed(() => checkboxGroup ? checkboxGroup.modelValue.value : props.modelValue)
+  const model = computed({
+    get() {
+      return isGroup.value ? store.value : props.modelValue !== undefined ? props.modelValue : selfModel
+    },
+
+    set(val: unknown) {
+      if (isGroup.value && Array.isArray(val)) {
+        isLimitExceeded.value = false
+
+        if (checkboxGroup.min !== undefined && val.length < checkboxGroup.min) {
+          isLimitExceeded.value = true
+        }
+        if (checkboxGroup.max !== undefined && val.length > checkboxGroup.max) {
+          isLimitExceeded.value = true
+        }
+
+        isLimitExceeded.value === false && checkboxGroup.changeEvent?.(val)
+      } else {
+        emit(UPDATE_MODEL_EVENT, val)
+        selfModel = val as boolean
+      }
+    },
+  })
+
+  return {
+    model,
+    isLimitExceeded,
+  }
+}
+
+const useCheckboxStatus = (props: ICheckboxProps, { model }: IArgs<typeof useModel>) => {
+  const { isGroup, checkboxGroup, elFormItemSize, ELEMENT } = useCheckboxGroup()
+  const focus = ref(false)
+  const size = computed<string|undefined>(() => checkboxGroup.checkboxGroupSize || elFormItemSize || (ELEMENT || {}).size)
+  const isChecked = computed(() => {
+    const value = model.value
+    if (toTypeString(value) === '[object Boolean]') {
+      return Boolean(value)
+    } else if (Array.isArray(value)) {
+      return value.includes(props.label)
+    } else if (value !== null && value !== undefined) {
+      return value === props.trueLabel
+    }
+  })
+  const checkboxSize = computed(() => {
+    const temCheckboxSize = props.size || elFormItemSize.value || (ELEMENT || {} as any).size
+    return isGroup.value
+      ? checkboxGroup.checkboxGroupSize || temCheckboxSize
+      : temCheckboxSize
+  })
+
+  return {
+    isChecked,
+    focus,
+    size,
+    checkboxSize,
+  }
+}
+
+const useDisabled = (
+  props: ICheckboxProps,
+  { model, isChecked }: IArgs<typeof useModel> & IArgs<typeof useCheckboxStatus>,
+) => {
+  const { elForm, isGroup, checkboxGroup } = useCheckboxGroup()
+  const isLimitDisabled = computed(() => {
+    const max = checkboxGroup.max
+    const min = checkboxGroup.min
+    return !!(max || min) && (model.value.length >= max && !isChecked.value) ||
+      (model.value.length <= min && isChecked.value)
+  })
+  const isDisabled = computed(() => {
+    return isGroup.value
+      ? checkboxGroup.disabled || props.disabled ||  (elForm as any || {} as any).disabled || isLimitDisabled.value
```

Comment:
Is `elForm` can asigned a type below? And also `elFormItem`.
```
type ElForm = {
  size?: string
  disabled?: boolean
}
```

I think this way is better than `as any`.

Reply:
This type will be provided by the `el-form`.

Reply:
YES, I also think so, you can wait for ElForm's update.